### PR TITLE
Rework to support quick and frequent stop/start of triggers with properties in Multibranch jobs

### PR DESCRIPTION
**Pull request by scoheb**

### CODE REVIEW

File: src/main/java/com/redhat/jenkins/plugins/ci/messaging/FedMsgMessagingWorker.java [hunk 1]
```diff
@@ -525,6 +529,18 @@ public String waitForMessage(Run<?, ?> build, TaskListener listener,
...
+            while (!pollerClosed) {
+                if (!Thread.currentThread().isAlive()) {
+                    log.info("poller not closed yet BUT trigger thread is dead. continuing interrupt");
+                    break;
+                }
+                try {
+                    log.info("poller not closed yet");
+                    Thread.currentThread().sleep(1000);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/messaging/FedMsgMessagingWorker.java]: note to myself: clean this up!

File: src/main/java/com/redhat/jenkins/plugins/ci/messaging/FedMsgMessagingWorker.java [hunk 2]
```diff
@@ -525,6 +529,18 @@ public String waitForMessage(Run<?, ?> build, TaskListener listener,
     public void prepareForInterrupt() {
         interrupt = true;
         try {
+            while (!pollerClosed) {
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/messaging/FedMsgMessagingWorker.java]: add max time to wait?

File: src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java [hunk 3]
```diff
@@ -175,63 +180,75 @@ private void startTriggerThread() {
...
 		}
 	}
 
-    private void stopTriggerThread() {
-        if (job != null) {
-            stopTriggerThread(job.getFullName());
-        }
-    }
+    private CITriggerThread stopTriggerThread(String fullName) {
+		synchronized(getLock(fullName)) {
```
Comment (ggallen) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java]: I would add a comment here indicating that the thread is created *only* for comparison purposes, and it is never started. This confused me at first.

Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java]: Done

**Pull request merged**